internal/mode/static/state/changed_predicate.go

```diff
@@ -31,17 +31,7 @@ type generationChangedPredicate struct{}
 
 func (generationChangedPredicate) delete(_ client.Object) bool { return true }
 
-func (generationChangedPredicate) upsert(oldObject, newObject client.Object) bool {
-	if oldObject == nil {
-		return true
-	}
-
-	if newObject == nil {
-		panic("Cannot determine if generation has changed on upsert because new object is nil")
-	}
-
-	return newObject.GetGeneration() != oldObject.GetGeneration()
-}
+func (generationChangedPredicate) upsert(_, _ client.Object) bool { return true }
```

Comment:
I recommend changing the name of this predicate to reflect its behavior. Maybe something like `alwaysTruePredicate`. 

Alternatively, if this is merged after #1432, you can remove this predicate and change the logic of the `changeTrackingUpdater` to return `true` if there's no `stateChangedPredicate` registered for the resource.

Reply:
Updated https://github.com/nginxinc/nginx-gateway-fabric/pull/1422/commits/52063131b95c1b067b79021892ec8e035fc3cc54